arroyo/processing/strategies/run_task.py

```diff
@@ -21,21 +31,36 @@ def __init__(
     ) -> None:
         self.__function = function
         self.__next_step = next_step
+        # Invalid message offsets pending commit
+        self.__invalid_messages: MutableSequence[InvalidMessage] = []
+
+    def __forward_invalid_offsets(self) -> None:
+        if self.__invalid_messages:
+            committable = {m.partition: m.offset + 1 for m in self.__invalid_messages}
```

Comment:
same comments as in `decoder`. also should we deduplicate this code somehow? the details of how this works are quite hard to get right from where I'm sitting, though the loc is low